test/test_sys_checkout.py

```diff
@@ -819,10 +819,15 @@ def _check_container_component_post_checkout(self, overall, tree):
 
     def _check_container_component_post_checkout2(self, overall, tree):
         self.assertEqual(overall, 0)
-        self._check_simple_opt_ok(tree)
         self._check_simple_tag_empty(tree)
         self._check_simple_branch_ok(tree)
 
+    def _check_container_component_post_checkout3(self, overall, tree):
+        self.assertEqual(overall, 0)
+        self.assertFalse("simp_opt" in tree)
+        self._check_simple_tag_ok(tree)
+        self._check_simple_branch_ok(tree)
```

Comment:
You should not expect this status entry to be 'clean' because it is not there. These two lines should be:
```
        self._check_simple_tag_empty(tree)
        self._check_simple_branch_empty(tree)
```

Reply:
This is incorrect - the only thing that I excluded is simple_opt - simple_tag and simple_branch are checked out.   The problem seems to be that these are marked as:
```
 ../../../checkout_externals -S -e externals.cfg -x simp_opt -v
Processing externals description file : externals.cfg
Processing externals description file : sub-externals.cfg
Checking status of externals: simp_branch, simp_tag, mixed_req, simp_branch, 
s   ./externals/mixed_req
        clean sandbox, 6fc379457ecb4e576a13c7610ae1fa73f845ee6a --> unknown_remote/master
s   ./externals/mixed_req/src/simp_branch
        clean sandbox, 36418b4e5665956a90725c9a1b5a8e551c5f3d48 --> unknown_remote/feature2
s   ./externals/simp_branch
        clean sandbox, 36418b4e5665956a90725c9a1b5a8e551c5f3d48 --> unknown_remote/feature2
    ./externals/simp_tag
        clean sandbox, on tag1
```
I don't understand why these are marked and in fact if I do another checkout:
```
 ../../../checkout_externals  -e externals.cfg -x simp_opt -v
Processing externals description file : externals.cfg
Processing externals description file : sub-externals.cfg
Checking status of externals: simp_branch, simp_tag, mixed_req, simp_branch, 
Checking out externals: 

s   ./externals/simp_branch
        clean sandbox, 36418b4e5665956a90725c9a1b5a8e551c5f3d48 --> unknown_remote/feature2
    git checkout --quiet repos_simple-ext.git/feature2

    ./externals/simp_tag
        clean sandbox, on tag1

s   ./externals/mixed_req
        clean sandbox, 6fc379457ecb4e576a13c7610ae1fa73f845ee6a --> unknown_remote/master
    git checkout --quiet repos_mixed-cont-ext.git/master

Processing externals description file : sub-externals.cfg
Checking out externals: 

--- 
        -,  --> 
    git checkout --quiet repos_simple-ext.git/feature2
```
I get the same hashes but now they are marked as clean:
```
cheyenne5: ~/sandboxes/cesm2_x_alpha/me.git/test/tmp/test_container_exclude_component
:) ../../../checkout_externals -S -e externals.cfg -x simp_opt -v
Processing externals description file : externals.cfg
Processing externals description file : sub-externals.cfg
Checking status of externals: simp_branch, simp_tag, mixed_req, simp_branch, 
    ./externals/mixed_req
        clean sandbox, on 6fc379457ecb4e576a13c7610ae1fa73f845ee6a
    ./externals/mixed_req/src/simp_branch
        clean sandbox, on 36418b4e5665956a90725c9a1b5a8e551c5f3d48
    ./externals/simp_branch
        clean sandbox, on 36418b4e5665956a90725c9a1b5a8e551c5f3d48
    ./externals/simp_tag
        clean sandbox, on tag1
```